src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -383,100 +397,281 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
+            //     M(a, /* a */
+            //       b);
+            // 
+            // However, in this case, we want the comment to be associated with node 'b':
+            //     M(a, /* b */ b);
+
+            var newParameters = list.ToList();
+            var newSeparators = new SyntaxToken[list.SeparatorCount];
+
+            for (var separatorIndex = 0; separatorIndex < list.SeparatorCount; separatorIndex++)
+            {
+                var separator = list.GetSeparator(separatorIndex);
+
+                // If the separator's trailing trivia does not contain any end-of-line trivia, we append the trailing trivia to the leading trivia of the node following.
+                // Otherwise, we leave the separator as is.
+                if (!separator.TrailingTrivia.Any(trivia => trivia.IsKind(SyntaxKind.EndOfLineTrivia)))
+                {
+                    newParameters[separatorIndex + 1] = newParameters[separatorIndex + 1].WithPrependedLeadingTrivia(separator.TrailingTrivia);
+                    separator = separator.WithoutTrailingTrivia();
+                }
 
-                newParameters.Add(param);
+                newSeparators[separatorIndex] = separator;
             }
 
-            var numSeparatorsToSkip = originalParameters.Count - reorderedParameters.Count;
-            return SyntaxFactory.SeparatedList(newParameters, GetSeparators(list, numSeparatorsToSkip));
+            return SyntaxFactory.SeparatedList(newParameters, newSeparators);
         }
 
-        private static T TransferLeadingWhitespaceTrivia<T>(T newArgument, SyntaxNode oldArgument) where T : SyntaxNode
+        private static (T newParamNode, SyntaxToken newSeparator) TransferTrivia<T>(
+            SeparatedSyntaxList<T> list,
+            T newParamNode,
+            int originalIndex,
+            int newIndex,
+            int reorderedParametersCount) where T : SyntaxNode
         {
-            var oldTrivia = oldArgument.GetLeadingTrivia();
-            var oldOnlyHasWhitespaceTrivia = oldTrivia.All(t => t.IsKind(SyntaxKind.WhitespaceTrivia));
+            // Copy whitespace trivia from original position.
+            newParamNode = TransferLeadingWhitespaceTrivia(newParamNode, list[newIndex]);
+
+            // The separator associated with the node should have its trivia transferred over as well. If we're looking at the last node, then there is no associated separator.
+            var originalSeparator = originalIndex < list.Count - 1 ? list.GetSeparator(originalIndex) : default;
+            var newSeparator = newIndex < reorderedParametersCount - 1 ? list.GetSeparator(newIndex) : default;
+
+            // If the associated node is not switching positions, we don't need to do any work.
+            if (originalSeparator == newSeparator)
+            {
+                return (newParamNode, newSeparator);
+            }
+
+            // If we have a comment of format '//', we switch the trivia notation to /* */.
```

Comment:
needs a 'why'.  like "because we could be putting the comment before a parameter, and we don't want it to now be commented out.